### ISSUE TITLE
[Feature Request] Repeat 1 and Repeat All

### ISSUE THREAD

**Issue by antidaeophobic** [HardcodeCoder/PulseMusic#109]
The repeat button, from what I can gather, is a "repeat 1" button. The icon could be changed to reflect this.
Additionally, the button could have a second function of replaying the playlist once it is over, instead of the existing behaviour of simply stopping playback.

I request this because I have a playlist called "on repeat," consisting of 5-6 songs that I listen to, well, on repeat. Manually restarting the playlist every 20 minutes isn't convenient.

Other music apps like Spotify have the repeat button functioning as follows:
Default: no repeat
First tap: repeat all
Second tap: repeat 1 (icon changes to contain a small "1")
Third tap: return to default

![replay.png](https://user-images.githubusercontent.com/62526737/115846299-a982ee80-a419-11eb-872e-cef78c4f0d6a.png)![replay2.png](https://user-images.githubusercontent.com/62526737/115846325-b1db2980-a419-11eb-90fe-b5bda7cdf805.png)![replay3.png](https://user-images.githubusercontent.com/62526737/115846346-b869a100-a419-11eb-8acd-58199761f5b1.png)

**Comment by HardcodeCoder** [comment]
Yes, I agree we can add more functionality to the repeat button. The current implementation is however not a "repeat 1" but a "repeat indefinitely". If you enable repeat it will continue looping the track forever unless you manually skip track. In order to remove confusion we can add a second repeat button in playlist details page ( Playing queue) that can loop the playlist instead.
